WinImageBuilder.psm1

```diff
@@ -848,6 +938,65 @@ function New-MaaSImage {
+    <#
+    .SYNOPSIS
+     This function generates a Windows image using Hyper-V as hypervisor.
+    .DESCRIPTION
+     This command requires Hyper-V to be enabled, a VMSwitch to be configured for external
+     network connectivity if the updates are to be installed, which is highly recommended.
+     This command uses internally the New-WindowsCloudImage to generate the base image and
+     start a Hyper-V instance using the base image. After the Hyper-V instance shuts down, 
+     the resulting VHDX is shrunken to a minimum size and converted to the required format.
+    .PARAMETER WimFilePath
+     The location of the WIM file from the Windows ISO.
+    .PARAMETER ImageName
+     This is the object which contains all the information about the Windows
+     flavor selected.
+    .PARAMETER WindowsImagePath
+     The destination of the generated image.
+    .PARAMETER SizeBytes
+     The size of the VHD used to generate the image.
+    .PARAMETER DiskLayout
+     This parameter can be set to either BIOS or UEFI.
+    .PARAMETER ProductKey
+     The product key for the OS selected.
+    .PARAMETER VirtIOISOPath
+     The path to the ISO file containing the VirtIO drivers.
+    .PARAMETER InstallUpdates
+     If set to true, the latest updates will be downloaded and installed.
+    .PARAMETER AdministratorPassword
+     Is used by the script to connect to the instance while it is generating.
+    .PARAMETER ExtraFeatures
+     Name of the extra features to enable on the generated image.
+     The $ExtraFeatures need be present in the ISO file.
+    .PARAMETER ExtraDriversPath
+     The path to the additional drivers to install.
+    .PARAMETER PersistDriverInstall
+     In case the hardware on which the image is generated will also be the hardware on
+     which the image will be deployed this can be set to true.
+    .PARAMETER Memory
+     RAM assigned to the VM used to generate the image.
+    .PARAMETER CpuCores
+     The number of CPU cores assigned to the VM used to generate the image.
+    .PARAMETER RunSysprep
+     Used to clean the OS on the VM, and to prepare it for a first-time use.
+    .PARAMETER SwitchName
+     Used to specify the virtual switch the VM will be using to connect to the internet.
+     If none is specified, one will be created.
+    .PARAMETER Force
+     It will force the image generation when $RunSysprep is $False or the selected $SwitchName 
+     is not an external one. Use this parameter with caution because it can easily generate
+     unstable images.
+    .PARAMETER Type
+     This parameter allows to choose between MaaS, KVM and Hyper-V specific images.
+    .PARAMETER PurgeUpdates
+     If set to true, will run DISM with /resetbase option. This will reduce the size of 
+     WinSXS folder, but after that Windows updates will not be possible to uninstall.
+    .PARAMETER DisableSwap
+     If set to $True it will disable the swap before the Sysprep process starts. This way the issues

```

Comment:
wrong description

DisableSwap option will disable the swap when the image is generated and will add a setting in the Unattend.xml file which will enable swap at boot time during specialize step. This is required as by default, the amount of swap space on Windows machine is directly proportional to the RAM size and if the image has in the initial stage low disk space, the first boot will fail due to not enough disk space. The swap is set to the default automatic setting right after the resize of the partitions is performed by cloudbase-init.